pkg/app/piped/cmd/piped/piped.go

```diff
@@ -377,6 +385,31 @@ func (p *piped) loadConfig() (*config.PipedSpec, error) {
 	return cfg.PipedSpec, nil
 }
 
+func (p *piped) initializeSealedSecretDecrypter(cfg *config.PipedSpec) (crypto.Decrypter, error) {
+	ssm := cfg.SealedSecretManagement
+	if ssm == nil {
+		return nil, nil
+	}
+
+	switch ssm.Type {
+	case model.SealedSecretManagementNone:
+		return nil, nil
+
+	case model.SealedSecretManagementSealingKey:
+		if ssm.PrivateKeyFile == "" {
+			return nil, fmt.Errorf("sealedSecretManagement.privateKeyFile must be set")
+		}
+		decrypter, err := crypto.NewRSADecrypter(ssm.PrivateKeyFile)
+		if err != nil {
+			return nil, fmt.Errorf("failed to initialize decrypter (%w)", err)
+		}
+		return decrypter, nil
+
```

Comment:
nit, so it's up to you.
```suggestion
	case model.SealedSecretManagementGCPKMS:
		return nil, fmt.Errorf("type %q is not implemented yet", ssm.Type.String())

	case model.SealedSecretManagementAWSKMS:
		return nil, fmt.Errorf("type %q is not implemented yet", ssm.Type.String())

```

Reply:
Thank you.